internal_transfer_with_agreed_amount/wizards/internal_transfer_multicurrency.py

```diff
@@ -64,8 +70,10 @@ def apply(self):
             amls.move_id.button_draft()
             debit = amls.filtered(lambda x: x.debit > 0)
             credit = amls.filtered(lambda x: x.credit > 0)
-            debit.with_context(check_move_validity=False).write({"debit": self.agreed_amount})
-            credit.with_context(check_move_validity=False).write({"credit": self.agreed_amount})
+            vals = self._prepare_values_from_lines(debit, is_same_company_currency=True)
+            debit.with_context(check_move_validity=False).write(vals)
+            vals = self._prepare_values_from_lines(credit, is_same_company_currency=True)
+            credit.with_context(check_move_validity=False).write(vals)
             amls.move_id.action_post()
             to_reconcile.reconcile()
             return
```

Comment:
Sorry, you have to cal the method before the return too  self._postprocessing_payment_with_agreed_amount(payment)

Reply:
Added.